pkg/nats/stream/consumer_client.go

```diff
@@ -293,16 +289,20 @@ func (nc *ConsumerClient) heartBeatRespHandler(msg *nats.Msg) {
 func (nc *ConsumerClient) createNewRequestAndSend(
 	ctx context.Context,
 	subj string,
-	producerConnID string,
 	data []byte) (*streamingBucket, error) {
 	nc.mu.Lock()
 
 	// Create new literal Inbox and map to a bucket.
 	respInbox := nc.newRespInbox()
 	token := respInbox[nc.respSubLen:]
-	bucket := newStreamingBucket(ctx, token, producerConnID)
-
+	bucket := newStreamingBucket(ctx, token, subj)
 	nc.respMap[token] = bucket
+
+	// Ensure map for this subject exists
+	if _, ok := nc.reqSubMap[subj]; !ok {
+		nc.reqSubMap[subj] = make(map[string]*streamingBucket)
+	}
+	nc.reqSubMap[subj][token] = bucket
```

Comment:
nit: I think this can be a map from subject to buckets or tokens. Not sure if a dimensional map is helping here